src/Adapter/Chain/CachePoolChain.php

```diff
@@ -107,19 +108,19 @@ public function getItems(array $keys = [])
                 foreach ($items as $item) {
                     if ($item->isHit()) {
                         $hits[$item->getKey()] = $item;
+                        unset($keys[array_search($item->getKey(), $keys)]);
                     }
+                    $loadedItems[$item->getKey()] = $item;
                 }
-
-                if (count($hits) === count($keys)) {
+                if (count($hits) === $keysCount) {
                     return $hits;
                 }
             } catch (CachePoolException $e) {
                 $this->handleException($poolKey, __FUNCTION__, $e);
             }
         }
 
-        // We need to accept that some items where not hits.
-        return array_merge($hits, $items);
+        return array_merge($hits, $loadedItems);
```

Comment:
But $hits contain Generator, not array. When you use return array_merge($loadedItems, $hits);?, you will got fatal